src/gen/conditions.rs

```diff
@@ -96,23 +96,35 @@ fn parse_args(
             c = rest(a, c)?;
             let message = sanitize_announce_msg(a, first(a, c)?, ErrorCode::InvalidMessage)?;
             // AGG_SIG_UNSAFE takes exactly two parameters
-            // but the argument list still doesn't need to be terminated by NIL,
-            // just any atom will do
-            match a.sexp(rest(a, c)?) {
-                SExp::Pair(_, _) => Err(ValidationErr(c, ErrorCode::InvalidCondition)),
-                _ => Ok(Condition::AggSigUnsafe(pubkey, message)),
+            if (flags & COND_ARGS_NIL) != 0 {
+                // make sure there aren't more than two
+                check_nil(a, rest(a, c)?)?;
+                Ok(Condition::AggSigUnsafe(pubkey, message))
+            } else {
+                // but the argument list still doesn't need to be terminated by NIL,
+                // just any atom will do
+                match a.sexp(rest(a, c)?) {
+                    SExp::Pair(_, _) => Err(ValidationErr(c, ErrorCode::InvalidCondition)),
+                    _ => Ok(Condition::AggSigUnsafe(pubkey, message)),
```

Comment:
Consider doing the `if flags` check here... seems like it might be a smaller change.

(Obviously not a critical change.)

Reply:
true. in my mind, the main reason to make this change is the opportunity to remove the existing logic down the line. So that's what I had in mind, what this code would look like with just the new rule